docs/src/main/asciidoc/spring-cloud-sleuth.adoc

```diff
@@ -191,7 +191,120 @@ adding a Channel Binder implementation
 (e.g. `spring-cloud-starter-stream-rabbit` for RabbitMQ or
 `spring-cloud-starter-stream-kafka` for Kafka). This will
 automatically turn your app into a producer of messages with payload
-type `Spans`. 
+type `Spans`.
+
+== Customizations
+
+Thanks to the `SpanInjector` and `SpanExtractor` you can customize the way spans
+are created and propagated.
+
+Essentially with Spring Cloud Span there are two ways span tracing data can be passed to
+ a process.
+
+ * via Spring Integration
+ * via HTTP
+
+=== Spring Integration
+
+For Spring Integration these are the beans responsible for creation of a Span from a `Message`
+ and filling in the `MessageBuilder` with tracing information.
+
+[source,java]
+----
+	@Bean
+	@Qualifier("messagingSpanExtractor")
+	@ConditionalOnProperty(value = "spring.sleuth.integration.extractor.enabled", matchIfMissing = true)
+	public SpanExtractor<Message> messagingSpanExtractor() {
+		...
+	}
+
+	@Bean
+	@Qualifier("messagingSpanInjector")
+	@ConditionalOnProperty(value = "spring.sleuth.integration.injector.enabled", matchIfMissing = true)
+	public SpanInjector<MessageBuilder> messagingSpanInjector() {
+		...
+	}
+----
+
+As you can see you can disable them with the properties:
+
+* `Message` span extractor - `spring.sleuth.integration.extractor.enabled`
```

Comment:
interesting properties.. I think the idea is that you want to stop propagation, right? like ingress and egress?

these nobs say for example, you have a scenario where you want to take spans, but not pass them down and visa versa.

do we need a knob like this yet? or would it be simpler to just disable the feature? that way, the more jargony terms extractor injector don't need to leak into property namespace.